### ISSUE TITLE
anndata concatenation fails for self.unknown_celltype_label="unknown"

### ISSUE THREAD

**Issue by annashch-insitro** [YosefLab/PopV#27]
using anndata version = 
```
Name: anndata
Version: 0.8.0
Summary: Annotated data.
Home-page:
Author: Philipp Angerer, Alex Wolf, Isaac Virshup, Sergei Rybakov
Author-email:
License:
Location: /opt/miniconda3/lib/python3.10/site-packages
Requires: h5py, natsort, numpy, packaging, pandas, scipy
Required-by: mudata, OnClass, popv, scanpy, scvi-tools
```

Seeing anndata concatenation error https://github.com/YosefLab/PopV/blob/main/popv/preprocessing.py#L285 when `self.unknown_celltype_label = "unknown"`. Issue manifests as a type conversion problem -- using the Tabula sapiens Liver reference tissue. Issue goes away when `self.unknown_celltype_label=None`

**Comment by cane11** [comment]
It's hard to follow this issue. Can you please also upload the exact error? The concatenation is not the most stable thing as it was changed in the recent Scanpy version.

**Comment by cane11** [comment]
Closing due to inactivity and issue is not reproducible on my site.
